### ISSUE TITLE
Can't build a docker image based on alpine

### ISSUE THREAD

**Issue by RockyNiu** [taosdata/TDengine#75]
Dockerfile:
```
FROM alpine AS builder
RUN apk add --no-cache git\
    && apk add --no-cache build-base\
    && apk add --no-cache cmake
RUN git clone https://github.com/taosdata/TDengine.git
WORKDIR /TDengine
RUN mkdir build && cd build && cmake .. && cmake --build .
```

The error:
```
-- Use the iconv functions in GLIBC
-- build version 
CMake Error at src/client/CMakeLists.txt:31 (SET_TARGET_PROPERTIES):
  SET_TARGET_PROPERTIES called with incorrect number of arguments.

-- MVN is not installed and JDBC is not compiled
-- Configuring incomplete, errors occurred!
```

**Comment by francis-du** [comment]
#### @RockyNiu What`s your cmake version?

**Comment by RockyNiu** [comment]
`3.14.5-r0`

The following is the full logs
```
Sending build context to Docker daemon  13.31kB
Step 1/8 : FROM alpine AS builder
 ---> b7b28af77ffe
Step 2/8 : RUN apk add --no-cache git    && apk add build-base    && apk add cmake
 ---> Using cache
 ---> 78204044206a
Step 3/8 : RUN git clone https://github.com/taosdata/TDengine.git
 ---> Using cache
 ---> 766744c80eb7
Step 4/8 : WORKDIR /TDengine
 ---> Using cache
 ---> c584e2bbc2bf
Step 5/8 : RUN mkdir build && cd build && cmake .. && cmake --build .
 ---> Running in 0ddb1199213a
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Project source directory: /TDengine
-- Build Debug Version
-- Project binary output path: /TDengine/build
-- Executable files output path: /TDengine/build/build/bin
-- Project library output path: /TDengine/build/build/lib
-- Check targe platform x64, pass
-- Use the iconv functions in GLIBC
-- build version 
CMake Error at src/client/CMakeLists.txt:31 (SET_TARGET_PROPERTIES):
  SET_TARGET_PROPERTIES called with incorrect number of arguments.


-- MVN is not installed and JDBC is not compiled
-- Configuring incomplete, errors occurred!
See also "/TDengine/build/CMakeFiles/CMakeOutput.log".
The command '/bin/sh -c mkdir build && cd build && cmake .. && cmake --build .' returned a non-zero code: 1
```

**Comment by squ** [comment]
git config --global core.autocrlf false

**Comment by tedwuusa** [comment]
I ran into same issue. I think alpine, being extremely minimal, is missing some required libraries/functions. For example, I saw pthread being used by `taosd`, but it is not part of alpine. 

Use ubuntu instead (for both docker build stages) worked for me. The image is a little bigger at 72MB (64MB ubuntu + 8MB TDengine), but still small enough for all practical purposes. You can save a few MB if only `taosd` is copied over.

```
FROM ubuntu AS build-env

RUN apt-get update
RUN apt-get -y install git
RUN apt-get -y install build-essential
RUN apt-get -y install cmake

RUN git clone https://github.com/taosdata/TDengine.git
RUN mkdir build && cd build && cmake ../TDengine && cmake --build .

FROM ubuntu
WORKDIR /app
ADD ./taos.cfg /app
COPY --from=build-env /build/build/bin/* /app/
RUN mkdir -p /data/log
ENTRYPOINT ["./taosd", "-c", "."]
```

**Comment by hzcheng** [comment]
There is a [blog](https://blog.csdn.net/qishidiguadan/article/details/96284529) using Centos image to build and run TDengine. You can refer to it. Besides, official docker image is on the road. We also invite that you can contribute the docker image.

**Comment by xrdavies** [comment]
Still struggling with alpine, since TDengine is to IoT alpine is necessary distro to be supported.

With  ver1.6.4.3, TDengine is still not able to be built.
Since alpine is not using glibc but musl (the libc for alpine), while TDengine uses glibc.

It will take lots of code changes to support Alpine. Take care!

**Comment by xrdavies** [comment]
> There is a [blog](https://blog.csdn.net/qishidiguadan/article/details/96284529) using Centos image to build and run TDengine. You can refer to it. Besides, official docker image is on the road. We also invite that you can contribute the docker image.


To run with other distro is so easy that not so challenge compared to running on Alpine. Hope I can fix this issue soon.

**Comment by xrdavies** [comment]
Fixed in #971
